### PR TITLE
DOC: correct docstring examples (#3439)

### PR DESCRIPTION
**Pull request by ProsperousHeart**
xref #3439

### CODE REVIEW

File: pandas/core/reshape/reshape.py [hunk 1]
```diff
@@ -979,22 +981,25 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
...
-    0          0.531828          0.724455          0.322959          0.293714
-    1          0.634401          0.611024          0.361789          0.630976
-    2          0.849432          0.722443          0.228263          0.092105
-    \
+    ... # doctest: +NORMALIZE_WHITESPACE
+       A(quarterly)-2010  A(quarterly)-2011  B(quarterly)-2010  B(quarterly)-2011  \
+    0           0.548814           0.544883           0.437587           0.383442
+    1           0.715189           0.423655           0.891773           0.791725
+    2           0.602763           0.645894           0.963663           0.528895
+
```
Comment (TomAugspurger) [inline pandas/core/reshape/reshape.py]: I think you have to remove this line (even though it seems like it *should* be there, based on the output).

Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: aw man ... WTF?^^ lol

File: pandas/core/reshape/reshape.py [hunk 2]
```diff
@@ -1129,8 +1134,7 @@ def get_dummies(data, prefix=None, prefix_sep='_', dummy_na=False,
     1  0  1    0
     2  0  0    1
 
-    >>> df = pd.DataFrame({'A': ['a', 'b', 'a'], 'B': ['b', 'a', 'c'],
-                        'C': [1, 2, 3]})
+    >>> df = pd.DataFrame({'A': ['a', 'b', 'a'], 'B': ['b', 'a', 'c'], 'C': [1, 2, 3]})
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: This will give a linting error (we check for PEP8)

The way you can solve this is by adding `...` (then it should pass the doctests):

```
>>> df = pd.DataFrame({'A': ['a', 'b', 'a'], 'B': ['b', 'a', 'c'],
...                    'C': [1, 2, 3]})
```

File: pandas/core/reshape/reshape.py [hunk 3]
```diff
@@ -940,6 +941,7 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
     8      3      3  2.1  2.9
     >>> l = pd.wide_to_long(df, stubnames='ht', i=['famid', 'birth'], j='age')
     >>> l
+    ... # doctest: +NORMALIZE_WHITESPACE
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: can you give some explanation why this is needed in this case ?

File: pandas/core/reshape/reshape.py [hunk 4]
```diff
@@ -689,7 +689,7 @@ def _convert_level_number(level_num, columns):
         new_labels = [np.arange(N).repeat(levsize)]
         new_names = [this.index.name]  # something better?
 
-    new_levels.append(level_vals)
+    new_levels.append(frame.columns.levels[level_num])
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: This is an actual change in the code. Is this to fix a bug? If so, could you do this as a separate PR (and add a test for it)?

File: pandas/core/reshape/reshape.py [hunk 5]
```diff
@@ -50,21 +50,21 @@ class _Unstacker(object):
...
-    two  3  4
+           a    b
+    one  1.0  2.0
+    two  3.0  4.0
 
     >>> s.unstack(level=0)
        one  two
-    a  1   2
-    b  3   4
+    a  1.0  3.0
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: Ah, this was actually an error in the example!

File: pandas/core/reshape/reshape.py [hunk 6]
```diff
@@ -50,21 +50,21 @@ class _Unstacker(object):
     ...                                    ('two', 'a'), ('two', 'b')])
     >>> s = pd.Series(np.arange(1.0, 5.0), index=index)
     >>> s
-    one  a   1
-         b   2
-    two  a   3
-         b   4
+    one  a    1.0
+         b    2.0
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: another option would be to change the series construction, use `np.arange(1, 5)` instead of `np.arange(1.0, 5.0)`.

I think using integers makes the example slightly simpler

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for working on this!

File: pandas/core/reshape/reshape.py (hunk 2 above)
Comment (TomAugspurger) [inline pandas/core/reshape/reshape.py]: Specifically, the lines should be less than 80 characters wide.

File: pandas/core/reshape/reshape.py (hunk 3 above)
Comment (TomAugspurger) [inline pandas/core/reshape/reshape.py]: @jorisvandenbossche I think the issue was when there's a `df.index.name` the output has whitespace for the rest of that line  (which we don't want to include in the source)

File: pandas/core/reshape/reshape.py (hunk 6 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: Agreed - working on this now.

Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: done - will be committing soon

File: pandas/core/reshape/reshape.py [hunk 7]
```diff
@@ -979,22 +981,25 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
...
+    
        X  id
     0  0   0
     1  1   1
-    2  2   2
+    2  1   2
+    
     >>> pd.wide_to_long(df, ['A(quarterly)', 'B(quarterly)'],
                         i='id', j='year', sep='-')
              X     A(quarterly)  B(quarterly)
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: Is it possible that the X column below is not correct? (it has no 1's)

File: pandas/core/reshape/reshape.py (hunk 5 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: Oh no! So ...... how do I fix it? Cause when I try to change it, I get an error

File: pandas/core/reshape/reshape.py (hunk 4 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: I didn't make that change so I don't know what that is

File: pandas/core/reshape/reshape.py (hunk 2 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: aw man .... I did that to several ...any suggested way to fix it?

File: pandas/core/reshape/reshape.py (hunk 3 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: Aha, ... 
Is that actually something we might want to solve in pandas, in the repr? (it's not a bug, but I also don't think it is a feature someone relies upon? So could change that (if it is easy) to not have to do this here) But that is certainly for another PR

File: pandas/core/reshape/reshape.py (hunk 2 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: like in mass? lol

File: pandas/core/reshape/reshape.py (hunk 7 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: X is random ... which didn't make sense to me.
`    >>> np.random.seed(0)
    >>> df = pd.DataFrame({'A(quarterly)-2010': np.random.rand(3),
    ...                    'A(quarterly)-2011': np.random.rand(3),
    ...                    'B(quarterly)-2010': np.random.rand(3),
    ...                    'B(quarterly)-2011': np.random.rand(3),
    ...                    'X' : np.random.randint(3, size=3)})
    >>> df['id'] = df.index
    >>> df
    ... # doctest: +NORMALIZE_WHITESPACE
       A(quarterly)-2010  A(quarterly)-2011  B(quarterly)-2010  B(quarterly)-2011  \
    0           0.548814           0.544883           0.437587           0.383442
    1           0.715189           0.423655           0.891773           0.791725
    2           0.602763           0.645894           0.963663           0.528895
       X  id
    0  0   0
    1  1   1
    2  1   2`

File: pandas/core/reshape/reshape.py (hunk 5 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: I think I fixed it?

File: pandas/core/reshape/reshape.py (hunk 3 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: Or we could set this flag globally if this occurs a lot? (if that is possible)

File: pandas/core/reshape/reshape.py [hunk 8]
```diff
@@ -979,22 +981,25 @@ def wide_to_long(df, stubnames, i, j, sep="", suffix='\d+'):
...
     ...                    'B(quarterly)-2011': np.random.rand(3),
     ...                    'X' : np.random.randint(3, size=3)})
     >>> df['id'] = df.index
     >>> df
-      A(quarterly)-2010 A(quarterly)-2011 B(quarterly)-2010 B(quarterly)-2011
-    0          0.531828          0.724455          0.322959          0.293714
-    1          0.634401          0.611024          0.361789          0.630976
-    2          0.849432          0.722443          0.228263          0.092105
-    \
+    ... # doctest: +NORMALIZE_WHITESPACE
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: You can also put this on the previous line (like the first example in https://docs.python.org/2/library/doctest.html#directives)

File: pandas/core/reshape/reshape.py (hunk 3 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: That's what @TomAugspurger was thinking, but I don't think he knows either?

File: pandas/core/reshape/reshape.py (hunk 5 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: No, no, now it is correct! So your change is perfect. I just noticed that by running the doctests, we actually corrected some errors in the docs, which is its purpose, so that is good :-)

File: pandas/core/reshape/reshape.py (hunk 4 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: Since it is in your commit, you somehow made that change :-) 
But if it was not the intent, you can just revert it (change it back to how it was based on the diff you see here)

File: pandas/core/reshape/reshape.py (hunk 8 above)
Comment (TomAugspurger) [inline pandas/core/reshape/reshape.py]: I think it was too long then :)

File: pandas/core/reshape/reshape.py (hunk 7 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: You can either change X to something not random, or leave it as is(with the random seed, it is also consistent), but the output below should just match the input (which is now not the case I think)

File: pandas/core/reshape/reshape.py [hunk 9]
```diff
@@ -48,23 +48,23 @@ class _Unstacker(object):
...
-    one  a   1
-         b   2
-    two  a   3
-         b   4
-    dtype: float64
+    one  a    1
+         b    2
+    two  a    3
+         b    4
+    dtype: int32
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: this one will depend on the platform you are using, so to be more robust, we should probably best add the dtype to `np.arange`. I would use 'int64', as this is the default in pandas (in numpy the default is platform dependent). 
Or as an alternative use `range(1, 5)`, then pandas Series will always convert this to int64

File: pandas/core/reshape/reshape.py (hunk 4 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: I didn't - I swear I didn't. I don't even know what that means. I'll change it back, but I promise it wasn't me.

File: pandas/core/reshape/reshape.py (hunk 8 above)
Comment (jorisvandenbossche) [inline pandas/core/reshape/reshape.py]: I meant

```
>>> df # doctest: +NORMALIZE_WHITESPACE
```

which would not be too long in this case

Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: ok - I'll add in next commit

File: pandas/core/reshape/reshape.py (hunk 9 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/reshape.py]: As per our discussion, left at int32 - np.arrange and range() did not seem to make a difference so I kept it at np.arrange

**Comment by ProsperousHeart** [comment]
What does it mean when the continuous-integration/travis-ci/py failed?

**Comment by jorisvandenbossche** [comment]
In this case it is due to some linting errors (some style issues, based on PEP8 and flake are checked in the third build on travis, which is the one that is failing. If you go to travis and click on the failing build, and scroll down, you can see why). 
The output from travis:

```
$ ci/lint.sh

inside ci/lint.sh
Linting  *.py
pandas/core/reshape/reshape.py:992:80: E501 line too long (84 > 79 characters)
pandas/core/reshape/reshape.py:993:80: E501 line too long (81 > 79 characters)
pandas/core/reshape/reshape.py:994:80: E501 line too long (81 > 79 characters)
pandas/core/reshape/reshape.py:995:80: E501 line too long (81 > 79 characters)
pandas/core/reshape/reshape.py:1000:1: W293 blank line contains whitespace
pandas/core/reshape/reshape.py:1001:80: E501 line too long (88 > 79 characters)
pandas/core/reshape/reshape.py:1015:80: E501 line too long (117 > 79 characters)
pandas/core/reshape/reshape.py:1133:71: W291 trailing whitespace
Linting *.py DONE
```

**Comment by jorisvandenbossche** [comment]
See http://pandas.pydata.org/pandas-docs/stable/contributing.html#python-pep8 (I would recommend you to set up the editor you are using to check for this as well, that should be possible with most code editors)

**Comment by ProsperousHeart** [comment]
I don't see any way (may not have search correct) for Notepad++ and PEP8

File: pandas/core/reshape/pivot.py [hunk 10]
```diff
@@ -438,27 +438,26 @@ def crosstab(index, columns, values=None, rownames=None, colnames=None,
...
-           shiny, dull, shiny, shiny, shiny], dtype=object)
+    >>> a = np.array(["foo", "foo", "foo", "foo", "bar", "bar",
+    ...               "bar", "bar", "foo", "foo", "foo"], dtype=object)
+    >>> b = np.array(["one", "one", "one", "two", "one", "one",
+    ...               "one", "two", "two", "two", "one"], dtype=object)
+    >>> c = np.array(["dull", "dull", "shiny", "dull", "dull", "shiny",
+    ...               "shiny", "dull", "shiny", "shiny", "shiny"], 
+    ...               dtype=object)
 
     >>> crosstab(a, [b, c], rownames=['a'], colnames=['b', 'c'])
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/pivot.py]: Can you make this `pd.crosstab` instead of `crosstab` ?

File: pandas/core/reshape/tile.py [hunk 11]
```diff
@@ -75,13 +75,11 @@ def cut(x, bins, right=True, labels=None, retbins=False, precision=3,
...
-      (6.533, 9.7], (0.191, 3.367]]
-    Categories (3, object): [(0.191, 3.367] < (3.367, 6.533] < (6.533, 9.7]],
-    array([ 0.1905    ,  3.36666667,  6.53333333,  9.7       ]))
+    ([(0.19, 3.367], (0.19, 3.367], (0.19, 3.367], (3.367, 6.533], (6.533, 9.7], (0.19, 3.367]]
+    Categories (3, interval[float64]): [(0.19, 3.367] < (3.367, 6.533] < (6.533, 9.7]], array([ 0.1905    ,  3.36666667,  6.53333333,  9.7       ]))
 
-    >>> pd.cut(np.array([.2, 1.4, 2.5, 6.2, 9.7, 2.1]), 3,
-               labels=["good","medium","bad"])
+    >>> result, bins = pd.cut(np.array([.2, 1.4, 2.5, 6.2, 9.7, 2.1]),
+    ...                                  3, labels=["good","medium","bad"])
```
Comment (jorisvandenbossche) [inline pandas/core/reshape/tile.py]: Here, you still would need 

```
>>> result
... [output]
>>> bins
... [output]
```

File: pandas/core/reshape/pivot.py (hunk 10 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/pivot.py]: done - will be in next commit

File: pandas/core/reshape/tile.py (hunk 11 above)
Comment (ProsperousHeart) [inline pandas/core/reshape/tile.py]: I'm not sure what that would look like :(

**Comment by ProsperousHeart** [comment]
Once confirmed, pivot completed.

**Comment by TomAugspurger** [comment]
Just pushed some fixes for the pep8 failures, and enabled the doctests on the doc build. We'll check the output of https://travis-ci.org/pandas-dev/pandas/jobs/236410485 when it finishes.

**Comment by jorisvandenbossche** [comment]
@ProsperousHeart Thanks a lot for working on this! 
Merging this now (follow-up PRs to do further improvements are always welcome!)

The appveyor failure is the matplotlib one, so ignoring that one.

There is one failing doctest, but that seems a python version issue (get different results when testing that locally as well on different python version). Will do another PR to check if changing doc build to py 3.6 solves things.
